pkg/pod/pod.go

```diff
@@ -371,19 +377,14 @@ func makeLabels(s *v1beta1.TaskRun) map[string]string {
 	return labels
 }
 
-// shouldAddReadyAnnotationOnPodCreate returns a bool indicating whether the
-// controller should add the `Ready` annotation when creating the Pod. We cannot
-// add the annotation if Tekton is running in a cluster with injected sidecars
-// or if the Task specifies any sidecars.
-func shouldAddReadyAnnotationOnPodCreate(ctx context.Context, sidecars []v1beta1.Sidecar) bool {
-	// If the TaskRun has sidecars, we cannot set the READY annotation early
-	if len(sidecars) > 0 {
-		return false
-	}
-	// If the TaskRun has no sidecars, check if we are running in a cluster where sidecars can be injected by other
-	// controllers.
-	cfg := config.FromContextOrDefaults(ctx)
-	return !cfg.FeatureFlags.RunningInEnvWithInjectedSidecars
+// isPodReadyImmediately returns a bool indicating whether the
+// controller should consider the Pod "Ready" as soon as it's deployed.
+// This will add the `Ready` annotation when creating the Pod,
+// and prevent the first step from waiting for the annotation to appear before starting.
+func isPodReadyImmediately(featureFlags config.FeatureFlags, sidecars []v1beta1.Sidecar) bool {
+	return !featureFlags.AwaitSidecarReadiness ||
```

Comment:
This expression is a bit hard to read, I think it's actually  a bit better when expanded into multiple statements